product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.1-converting-soap-to-json/1.1.1-soap-to-json-using-payloadfactory-mediator/src/test/java/org/wso2/carbon/esb/scenario/test/SoapToJsonUsingPayloadFactoryTest.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.esb.scenario.test;
+
+import org.apache.http.HttpResponse;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.wso2.carbon.esb.scenario.test.common.HttpConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioConstants;
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.esb.integration.common.utils.clients.SimpleHttpClient;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+public class SoapToJsonUsingPayloadFactoryTest extends ScenarioTestBase {
+
+    private String cappNameWithVersion = "approach_1_1_1_synapse_configCompositeApplication_1.0.0";
+    private String cappName = "approach_1_1_1_synapse_configCompositeApplication";
+    private String apiName = "1_1_1_soap_to_json_using_payload_factory";
+    private String apiInvocationUrl;
+
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+        apiInvocationUrl = getApiInvocationURLHttp(apiName);
+        deployCarbonApplication(cappNameWithVersion);
+    }
+
+    @Test(description = "1.1.1.1-Valid Soap To Json transformation", enabled = true, dataProvider = "1.1.1.1")
+    public void convertValidSoapToJson(String request, String expectedResponse, String header) throws Exception {
+        log.info("Executing test case 1.1.1.1");
+        log.info("apiInvocationUrl is set as : " + apiInvocationUrl);
+
+        SimpleHttpClient httpClient = new SimpleHttpClient();
+
+        Map<String, String> headers = new HashMap<String, String>();
+        headers.put(ScenarioConstants.MESSAGE_ID, header);
+        HttpResponse httpResponse = httpClient.doPost(apiInvocationUrl, headers, request, HttpConstants.MEDIA_TYPE_APPLICATION_XML);
+        String responsePayload = httpClient.getResponsePayload(httpResponse);
+
+        log.info("Actual response received 1.1.1.1: " + responsePayload);
+
+        Assert.assertEquals(httpResponse.getStatusLine().getStatusCode(), 200, "SOAP to JSON transformation failed");
+
+        Assert.assertEquals(expectedResponse.replaceAll("\\s+",""), responsePayload.replaceAll("\\s+",""));
```

Comment:
Shall we add assert fail message

Reply:
Shall I add the fail message to assertEquals?